diff --git a/HeroManStartKodeSkabelon.js b/HeroManDelopgave1.js
index 3a55a93..d633142 100644
--- a/HeroManStartKodeSkabelon.js
+++ b/HeroManDelopgave1.js
@@ -1,34 +1,37 @@
-// Start code template
+// HeroMan subtask 1
 // programming = 'Set your imagination free' / Jens Valdez
 // https://code-a-game.github.io
 
 let gameState;
 let heroMan;
 
 function setup() {
   createCanvas(600, 600);
 
   setStartValues();
 }
 
 function draw() {
   background(220);
 
   showGameState();
 }
 
 function setStartValues() {
   heroMan = new HeroMan();
 }
 
 function showGameState() {
   heroMan.show();
 }
 
 class HeroMan {
-  constructor() { }
+  constructor() {
+    this.x = 50;
+    this.y = 50;
+  }
 
   show() {
-    point(50, 50);
+    point(this.x, this.y);
   }
 }
