diff --git a/HeroManDelopgave3.js b/HeroManDelopgave4.js
index 1895769..331cd7d 100644
--- a/HeroManDelopgave3.js
+++ b/HeroManDelopgave4.js
@@ -1,61 +1,61 @@
-// HeroMan subtask 3
+// HeroMan subtask 4
 // programming = 'Set your imagination free' / Jens Valdez
 // https://code-a-game.github.io
 
 let gameState;
 let heroMan;
 
 function setup() {
   createCanvas(600, 600);
 
   setStartValues();
 }
 
 function draw() {
   background(220);
 
   showGameState();
   updateGameState();
 }
 
 function setStartValues() {
   heroMan = new HeroMan();
 }
 
 function showGameState() {
   heroMan.show();
 }
 
 function updateGameState() {
   heroMan.move();
 }
 
 class HeroMan {
   constructor() {
     this.x = 50;
     this.y = 50;
     this.radius = 25;
     this.colour = "yellow";
     this.moveSpeed = 2;
   }
 
   show() {
     fill(this.colour);
     circle(this.x, this.y, this.radius * 2);
   }
 
   move() {
-    if (keyIsDown(LEFT_ARROW)) {
+    if (keyIsDown(LEFT_ARROW) && this.x - this.radius > 0) {
       this.x -= this.moveSpeed;
     }
-    if (keyIsDown(RIGHT_ARROW)) {
+    if (keyIsDown(RIGHT_ARROW) && this.x + this.radius < width) {
       this.x += this.moveSpeed;
     }
-    if (keyIsDown(UP_ARROW)) {
+    if (keyIsDown(UP_ARROW) && this.y - this.radius > 0) {
       this.y -= this.moveSpeed;
     }
-    if (keyIsDown(DOWN_ARROW)) {
+    if (keyIsDown(DOWN_ARROW) && this.y + this.radius < height) {
       this.y += this.moveSpeed;
     }
   }
-}
+}
\ No newline at end of file
