diff --git a/zelda/zelda-code/ZeldaSubtask1.js b/zelda/zelda-code/ZeldaSubtask2.js
index 1234567..89abcde 100644
--- a/zelda/zelda-code/ZeldaSubtask1.js
+++ b/zelda/zelda-code/ZeldaSubtask2.js
@@ -1,15 +1,14 @@
-// ZeldaSubtask1 - Solution
+// ZeldaSubtask2
 // programming = 'Set your imagination free' / Jens Valdez
 // https://code-a-game.github.io
 
-function setup() {
-  createCanvas(400, 400);
-}
-
-function draw() {
-  background(220);
-}
+let hero;
 let gameMap;
 
 function preload() {
   imageBackground = loadImage(`images/background1.png`);
+  imageHero = loadImage(`images/zelda.png`);
 }
 function setup() {
   createCanvas(483, 313);
 
   gameMap = new GameMap();
+  hero = new Hero();
 }
 
 function draw() {
   image(imageBackground, 0, 0, 483, 313);
 
   gameMap.show();
+  hero.show();
 
   //  text(mouseX + " " + mouseY, mouseX, mouseY);
 }
@@ -38,5 +37,46 @@ class GameMap {
     }
   }
 }
 
+class Hero {
+  constructor() {
+    this.x = 5;
+    this.y = 6;
+  }
+
+  moveLeft() {
+    if (this.x > 0) {
+      this.x--;
+    }
+  }
+
+  moveRight() {
+    if (this.x < 6) {
+      this.x++;
+    }
+  }
+
+  moveUp() {
+    if (this.y > 0) {
+      this.y--;
+    }
+  }
+
+  moveDown() {
+    if (this.y < 6) {
+      this.y++;
+    }
+  }
+
+  show() {
+    image(imageHero, 37 * this.x, 27 + 37 * this.y);
+  }
+}
+
+function keyPressed() {
+  if (keyCode === LEFT_ARROW) {
+    hero.moveLeft();
+  } else if (keyCode === RIGHT_ARROW) {
+    hero.moveRight();
+  } else if (keyCode === UP_ARROW) {
+    hero.moveUp();
+  } else if (keyCode === DOWN_ARROW) {
+    hero.moveDown();
+  }
+}
+
